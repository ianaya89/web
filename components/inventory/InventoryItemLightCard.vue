<template>
  <div class="inventory-item-card-light w-full flex flex-wrap h-full relative">
    <div class="absolute top-0 right-0">
      <span v-if="secondaryAction" data-cy="inventory-item-remove" class="fas fa-times text-red cursor-pointer" @click="onRemoveClick()" />
    </div>
    <div class="w-full flex flex-col pt-6">
      <inventory-icon class="h-10 w-10 self-center" :category="item.category" />
    </div>
    <div class="w-full items-end">
      <h6 class="mt-3 leading-tight text-h7">
        {{ item.nameOverride || $t('app.marketplace.categories.' + categories.tertiary) }}
      </h6>
      <p class="text-grey text-p3 py-2 -mt-2 leading-none">
        {{ $t('app.marketplace.categories.' + $helpers.normalizeCategoryName(categories.secondary)) }}
      </p>
    </div>
  </div>
</template>

<script>
import InventoryIcon from '@/components/inventory/InventoryIcon'

export default {
  name: 'InventoryItemLightCard',

  components: {
    InventoryIcon
  },

  props: {
    item: {
      type: Object,
      required: true
    },
    secondaryAction: {
      type: Boolean,
      default: true
    }
  },

  computed: {
    categories() {
      const categories = this.item.category.path.split(' / ')

      return {
        main: categories[0],
        secondary: categories[1],
        tertiary: categories[2]
      }
    }
  },

  methods: {
    onRemoveClick() {
      this.$emit('item-remove', this.item)
    }
  }
}
</script>

<style lang="postcss" scoped>
.inventory-item-card-light {
  min-height: 9rem;
}
</style>
