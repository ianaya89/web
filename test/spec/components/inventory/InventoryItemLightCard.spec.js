import { shallowMount } from '@vue/test-utils'
import InventoryItemLightCard from '@/components/inventory/InventoryItemLightCard'

describe('InventoryItemLightCard', () => {
  it('renders InventoryItemLightCard', () => {
    const wrapper = shallowMount(InventoryItemLightCard, {
      propsData: {
        item: {
          nameOverride: 'Name',
          category: {
            path: 'some / very / long / path'
          }
        }
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('emits item-remove event', () => {
    const item = {
      nameOverride: 'Name',
      category: {
        path: 'some / very / long / path'
      }
    }

    const wrapper = shallowMount(InventoryItemLightCard, {
      propsData: {
        item
      }
    })

    wrapper.vm.onRemoveClick()

    expect(wrapper.emitted('item-remove')).toBeTruthy()
    expect(wrapper.emitted('item-remove')[0][0]).toEqual(item)
  })
})
