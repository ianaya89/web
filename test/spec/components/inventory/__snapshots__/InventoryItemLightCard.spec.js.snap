// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryItemLightCard renders InventoryItemLightCard 1`] = `
<div
  class="inventory-item-card-light w-full flex flex-wrap h-full relative"
>
  <div
    class="absolute top-0 right-0"
  >
    <span
      class="fas fa-times text-red cursor-pointer"
      data-cy="inventory-item-remove"
    />
  </div>
   
  <div
    class="w-full flex flex-col pt-6"
  >
    <inventory-icon-stub
      category="[object Object]"
      class="h-10 w-10 self-center"
    />
  </div>
   
  <div
    class="w-full items-end"
  >
    <h6
      class="mt-3 leading-tight text-h7"
    >
      
      Name
    
    </h6>
     
    <p
      class="text-grey text-p3 py-2 -mt-2 leading-none"
    >
      
      TRANSLATE:app.marketplace.categories.very
    
    </p>
  </div>
</div>
`;
